aws_lambda_powertools/event_handler/api_gateway.py

```diff
@@ -1588,6 +1594,15 @@ def get_openapi_schema(
 
         # Add routes to the OpenAPI schema
         for route in all_routes:
+
+            if route.security and not validate_openapi_security_parameters(
```

Comment:
In this case we will have `if` here or there, right? So, maybe it doesn't make much difference because we will always have if to check whether the route is `None` or not. I think the code and the intention of what we want to do will be more explicit now. WDYT?

Reply:
> in this case we will have `if` here or there, right?

You're 10000000% right. I assumed Python would not short-circuit and waste resources evaluating both.. but [IT DOES](https://docs.python.org/3/library/stdtypes.html#boolean-operations-and-or-not).

I'm gonna resolve it.

My personal preference is to always encapsulate validation within the validator to avoid surprises, but changing or not here doesn't lead to any bug :)